#ifndef MAINWINDOW_H
#define MAINWINDOW_H
#include <QtNetwork>
#include <QMainWindow>
#include <VulnerabilityDatabase.h>
#include <QListWidgetItem>
#include <QTextEdit>
QT_BEGIN_NAMESPACE
namespace Ui { class MainWindow; }
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();
    void ShowDevices();
    static QTextEdit* debug_edit;
    // добавил загрузку данных в оп, добавил поле для поиска, добавил структуру cve-package
    //void myMessageOutput(QtMsgType type, const QMessageLogContext &context, const QString &msg);
private slots:

    void on_add_button_clicked();

    void on_tabWidget_currentChanged(int index);

    void on_pushButton_clicked();

    void on_scan_button_clicked();


    void on_delete_device_button_clicked();



    void on_package_search_radioButton_pressed();

    void on_cve_search_radioButton_pressed();

    void on_search_lineEdit_textChanged(const QString &arg1);

    void on_report_listWidget_itemClicked(QListWidgetItem *item);

private:
    Ui::MainWindow *ui;
    QNetworkAccessManager *networkManager;
    V_Database database;

};
#endif // MAINWINDOW_H
